### PR TITLE
Add support for Groups

### PR DESCRIPTION
**Pull request by b6b**
This PR adds Group and GroupCollection models. It also creates a new Library base class that User and Group inherit from to get library messages.

### CODE REVIEW

**Review by lhwalsh — APPROVED** [review]
LGTM

File: lib/qualtrics_api/group_collection.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    end
+
+    private
+
+    def build_result(element)
+      QualtricsAPI::Group.new(element)
+    end
+
+    def list_endpoint
+      'groups'
```
Comment (arincon9) [inline lib/qualtrics_api/group_collection.rb]: double quotes?

**Review by arincon9 — APPROVED** [review]
LGTM

**Comment by b6b** [comment]
Thanks @lhwalsh @arincon9

**Pull request merged**